go/client/cmd_pgp_gen.go

```diff
@@ -68,6 +70,45 @@ func (v *CmdPGPGen) Run() (err error) {
 	return err
 }
 
+var CheckOptionalEmail = libkb.Checker{
+	F: func(s string) bool {
+		if len(s) == 0 {
+			return true
+		}
+		return libkb.CheckEmail.F(s)
+	},
+	Hint: libkb.CheckEmail.Hint,
+}
+
+func (v *CmdPGPGen) propmptPGPIDs() (err error) {
+	id := libkb.Identity{}
+	prompt := "Enter a public username for this key"
+	id.Username, err = PromptWithChecker(PromptDescriptorPGPGenEnterID, GlobUI, prompt, false, libkb.CheckUsername)
+	if err != nil {
+		return
+	}
+	prompt = "Enter a public email address for your key (or <enter> when done)"
```

Comment:
As I showed in the above comment, you can skip the email for the primary ID. Do we want to allow that, with better wording? Or should we insist on an email at that stage?